[19.0] [MIG] base_import_async: migrate

tishmen opened the pull request:
Scope:
- base_import_async only

Depends on:
- [x] https://github.com/OCA/queue/pull/840

Summary:
- Migrate to Odoo 19.

Temporary references:
- test-requirements.txt adds a temporary dependency to the base PR for CI/runboat. (This will be removed once the base PR merges.)

Pre-commit:
- Ran locally before opening; auto-fixes committed.

Tests:
- Local install may be skipped where core base_import is absent; CI on OCA infra will validate.

sbidoul commented:
/ocabot migration base_import_async

hoangtrann reviewed (approved):
Code LGTM, though you can now remove queue_job from test-requirements.txt

tishmen committed:
[19.0][MIG] base_import_async: migrate

- Move BaseImportImport wizard code to wizard/\n- Update tests and async import flow\n- Update pre-commit exclusions

tishmen commented:
Hi @hoangtrann @sbidoul , I rebased the branch on 19.0, resolved the conflicts, and addressed the review nits (removed the obsolete test-requirements.txt dependency and the “Odoo 19” comment). I also restored requirements.txt (generated from external_dependencies) and added tests to improve coverage. Finally, I squashed the PR into a single commit; CI is running again on the new head. Please double check if this is ready to get merged.

AnizR commented on base_import_async/wizard/base_import_import.py:
```suggestion
OPT_CHUNK_SIZE = "limit"
```

It had been fixed in 16.0: github.com/OCA/queue/pull/878

but it looks like it wasn't taken in previous migration

AnizR reviewed (changes requested):
Thanks for your contribution 🚀
